{{cookiecutter.project_name}}/pyproject.toml

```diff
@@ -1,132 +1,69 @@
-    "Programming Language :: Python :: 3.11",
-    "Programming Language :: Python :: 3.12",
-    "Programming Language :: Python :: 3.13",
-    "Topic :: Software Development :: Libraries :: Python Modules",
-]
+# See https://setuptools.readthedocs.io/en/latest/build_meta.html
+[build-system]
+requires = ["setuptools", "wheel"]
+build-backend = "setuptools.build_meta"
 
-[project.urls]
-Homepage = "https://{{cookiecutter.author_github_handle}}.github.io/{{cookiecutter.project_name}}/"
-Repository = "https://github.com/{{cookiecutter.author_github_handle}}/{{cookiecutter.project_name}}"
-Documentation = "https://{{cookiecutter.author_github_handle}}.github.io/{{cookiecutter.project_name}}/"
+[tool.ruff]
+line-length = 160
+target-version = "py311"
+lint.select = ["E", "F", "I"]
 
-[dependency-groups]
-dev = [
-    "pytest>=7.2.0",
-    "pre-commit>=2.20.0",
-    "tox-uv>=1.11.3",
-    {% if cookiecutter.deptry == 'y' %}"deptry>=0.22.0",{% endif %}
-    "mypy>=0.991",
-    {% if cookiecutter.codecov == 'y' %}"pytest-cov>=4.0.0",{% endif %}
-    "ruff>=0.9.2",
-    {% if cookiecutter.mkdocs == 'y' %}"mkdocs>=1.4.2",
-    "mkdocs-material>=8.5.10",
-    "mkdocstrings[python]>=0.26.1",{% endif %}
-]
+[tool.ruff.lint.per-file-ignores]
+"__init__.py" = ["F401"]
 
-[build-system]
-requires = ["hatchling"]
-build-backend = "hatchling.build"
+[tool.ruff.lint.isort]
+known-first-party = ["{{cookiecutter.project_slug}}"]
+
+[tool.pytest.ini_options]
+markers = ["requires_api_key"]
 
-{% if cookiecutter.layout == "flat" -%}
-[tool.setuptools]
-py-modules = ["{{cookiecutter.project_slug}}"]
-{%- endif %}
 
-[tool.mypy]
-files = [
-    {%- if cookiecutter.layout == "src" -%}
-    "src"
-    {%- else -%}
-    "{{cookiecutter.project_slug}}"
-    {%- endif -%}
+[tool.codeflash]
+# All paths are relative to this pyproject.toml's directory.
+module-root = "{{cookiecutter.project_slug}}"
+tests-root = "tests"
+test-framework = "pytest"
+ignore-paths = []
+formatter-cmds = ["ruff check --exit-zero --fix $file", "ruff format $file"]
+
+[project]
+name = "{{cookiecutter.project_name}}"
+version = "0.0.1"
+description = "{{cookiecutter.project_description}}"
+authors = [{ name = "{{cookiecutter.project_name}} authors", email = "{{cookiecutter.email}}" }]
+dependencies = [
 ]
-disallow_untyped_defs = true
-disallow_any_unimported = true
-no_implicit_optional = true
-check_untyped_defs = true
-warn_return_any = true
-warn_unused_ignores = true
-show_error_codes = true
+requires-python = ">=3.10"
+readme = "README.md"
+license = { text = "{{cookiecutter.open_source_license}}" }
 
-[tool.pytest.ini_options]
-testpaths = ["tests"]
+[project.optional-dependencies]
+optional_dependencies = []
+all = ["{{cookiecutter.project_name}}[visualization,dev,docs]"]
+dev = ["pytest", 'ruff', "pytest-cov", 'pytest-dotenv']
+docs = ["mkdocs", "mkdocs-material", "mkdocstrings[python]"]
+visualization = ["matplotlib>=3.8.2", "plotly>=5.18.0", "nomic>=1.0.0"]
```

Comment:
nomic can certainly go. 

I also do not know if we want to keep the rest here as a default